### DISCUSSION TITLE
Make GitHub Copilot aware of intellisense suggestions

### DISCUSSION THREAD

**Discussion by asportnoy** [community/community#42336] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Our codebase (JS) has a bunch of objects with various properties. When writing code that references those objects' properties, GitHub Copilot is just guessing on what property names exist on the object, and often gets them wrong. However, all the properties of the objects are documented in intellisense. So if GitHub Copilot would be able to access that information, that would result in much more accurate results.

```js
const SOME_OBJECT = /** @type {const} */ ({
    SOME_PROPERTY: 1,
    ANOTHER_PROPERTY: 2,
    THIRD_PROPERTY: 3,
    FOURTH_PROPERTY: 4
    UNRELATED_PROPERTY_NAME: 5, // Note this property
});

// This is often done in another file where Copilot is not aware of the code above

/** @type {Record<keyof typeof SOME_OBJECT, string>} */
const ANOTHER_OBJECT = ({
    [SOME_OBJECT.SOME_PROPERTY]: 1,
    [SOME_OBJECT.ANOTHER_PROPERTY]: 2,
    [SOME_OBJECT.THIRD_PROPERTY]: 3,
    [SOME_OBJECT.FOURTH_PROPERTY]: 4
    [SOME_OBJECT.FIFTH_PROPERTY]: 5, // GitHub Copilot suggested this line
});
```

**Comment by Awendel** [comment]
I second this
type information should definitely be added to the prompt, otherwise it performs worse than intellisense at the moment,
especially for languages that have strong types (such as typescript, ...)

It should really just forward type definitions to the prompt, not just variable names
